doc/api/cli.md

```diff
@@ -1028,6 +1028,12 @@ Valid values are `"commonjs"` and `"module"`. The default is `"commonjs"`.
 
 The REPL does not support this option.
 
+Additional details:
+
+* The default input type can be altered with
+  [`--experimental-default-type=module`][].
+* [`--print`][] does not support `--input-type=module`.
```

Comment:
Let's just remove the reference to `--print` from the earlier paragraph and then we don't need this bulleted list.

Reply:
I'm neutral on removing the mention of `--print` from the other paragraph, but it seems to me still interesting to explicitly say that `--print` does not work with ESM source – in other words, we need this bulleted list IMO.

Reply:
There could be an unlimited list of things this flag _doesn't_ apply to. We should just clean up the language that's already there to make clear that it applies to `--eval` and stdin and nothing else.

Reply:
`--input-type` definitely applies to `--print`, not sure what you mean

Reply:
I think its better to have this bullet point, let me know if this needs a change.

Reply:
> `--input-type` definitely applies to `--print`, not sure what you mean

“Applies” in the sense that it causes `--print` to throw an error?

```
node --input-type=module --print 'import.meta.url'
node:internal/process/execution:46
    throw new ERR_EVAL_ESM_CANNOT_PRINT();
    ^

Error [ERR_EVAL_ESM_CANNOT_PRINT]: --print cannot be used with ESM input
```

I don’t like the whole concept of an “Additional details” subsection. It’s too verbose. We should just modify the original text as minimally as possible to correct it:

```md
This configures Node.js to interpret `--eval` or `STDIN` input as CommonJS or as
an ES module. Valid values are `"commonjs"` or `"module"`. The default is
`"commonjs"` unless [`--experimental-default-type=module`][] is used.

The REPL does not support this option. Usage with [`--print`][] will throw an
error, as `--print` does not support ES module syntax.
```

Reply:
Or it causes `--print` to not throw an error:

```console
$ node --experimental-default-type=module --prrint 2
node:internal/process/execution:46
    throw new ERR_EVAL_ESM_CANNOT_PRINT();
    ^

Error [ERR_EVAL_ESM_CANNOT_PRINT]: --print cannot be used with ESM input
$ node --experimental-default-type=module --input-type=commonjs --prrint 2
2
```

Reply:
> Or it causes `--print` to not throw an error:

Can we just land the text I suggested in https://github.com/nodejs/node/pull/50796#discussion_r1405462690? I don’t see the point in debating `--print`.

Reply:
What's wrong with my text?

I feel strongly about keeping the docs as readable as possible. Two paragraphs plus a bulleted list is much too verbose for this.

Reply:
Okay, so if you wouldn't mind updating your branch with that text, then we can land this.